src/adventure_parser.rs

```diff
@@ -0,0 +1,73 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::path::Path;
+
+//rules of the syntax:
+//some character to signify the start of a choice
+//Narrator is the default character. It is reset to narrator at the start of each choice. A different character can be set by surrounding their name with back ticks: `
+//trailing '----' makes parsing a bit harder
+//each choice ends at an empty new line, and waits for user input
+//depending on user input, a choice is called
+//initial idea is to store choices in a map, so that you can easily switch between them. Aa grid like structure could be made to make a more logical structure, in which only potential choices are connected to the current one
+//need a character to signify an operation/function call
+
+struct Choice {
+    title: String,
+    dialogue: String,
+    option: String,
+}
+
+const TITLE: char = '#';
```

Comment:
Could the name here convey these are some sort of marker Rather than some singular title, speaker, or action?

Reply:
Do you have anything in mind?

Reply:
e.g. TITLE_MARKER, TITLE_TOKEN (nicer if we do a lexer later and can have an enum of possible tokens). Just so it's clear this is not the title itself.